### PR TITLE
add Trivia-QA dataset

### PR DESCRIPTION
**Pull request by sbmaruf**

### CODE REVIEW

**Comment by sbmaruf** [comment]
@VictorSanh  Please take a look here. I closed the previous pull.

File: templates/trivia_qa/rc/templates.yaml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    reference: Guess a question. It will show if model can evaluate entity in question.
+    task_template: false
+  69b1d0d7-ee22-4313-a8a0-89b718177d2a: !Template
+    id: 69b1d0d7-ee22-4313-a8a0-89b718177d2a
+    jinja: "{% if answer.type == \"WikipediaEntity\" %} \nI''ve always wondered: {{question}}{{\"\
+      \\n\"}}\nCan you search wikipedia and answer? ||| {{answer.aliases|choice}}\
+      \ \n{% endif %}\n"
+    name: context_self_description_wikipedia
+    reference: Ask question by self description from wikipedia
+    task_template: true
```
Comment (VictorSanh) [inline templates/trivia_qa/rc/templates.yaml:49]: it's a reading comprehension dataset
```suggestion
    task_template: false
```

File: templates/trivia_qa/rc/templates.yaml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    reference: Ask question using wikipedia keyword. Multiple answer.
+    task_template: true
+  ab0a59db-f3f8-4111-988d-235ab27f85da: !Template
+    id: ab0a59db-f3f8-4111-988d-235ab27f85da
+    jinja: "{% if answer.type == \"WikipediaEntity\" %} \nI''ve always wondered: {{question}}{{\"\
+      \\n\"}}\nCan you answer from Wikipedia? ||| {{answer.aliases|choice}} \n{% endif\
+      \ %}\n"
+    name: context_self_description_wikipedia_2
+    reference: Ask question by self description from wikipedia
+    task_template: true
```
Comment (VictorSanh) [inline templates/trivia_qa/rc/templates.yaml:79]: hmm quite confused about this one
the model won't be able to technically search Wikipedia.
You should transform this into `Here's what I found on the internet:... please answer the question`

Comment (VictorSanh) [reply templates/trivia_qa/rc/templates.yaml:79]: same for `context_self_description_wikipedia_2`

File: templates/trivia_qa/rc/templates.yaml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+subset: rc
+templates:
+  014c221a-ed54-4a1d-bc67-2ded6ea8c528: !Template
+    id: 014c221a-ed54-4a1d-bc67-2ded6ea8c528
+    jinja: "{% if answer.type == \"WikipediaEntity\" %} \nI've always wondered: {{question}}\
+      \ {{\"\\n\"}} Can you answer from wikipedia? You can try multiple times. |||\
+      \ {% for v in answer.aliases -%} Answer: {{v}} {% endfor -%}\n{% endif %}"
+    name: context_self_description_wikipedia_multiple_answer
+    reference: Ask question by self description from wikipedia
+    task_template: true
```
Comment (VictorSanh) [inline templates/trivia_qa/rc/templates.yaml:11]: you can remove this one, i am very not sure of the signal it brings and it doesn't make sense from the evaluation perspective

File: templates/trivia_qa/rc/templates.yaml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  2725b229-1217-43ed-98b0-32cf5c4293b3: !Template
+    id: 2725b229-1217-43ed-98b0-32cf5c4293b3
+    jinja: "{% if answer.type == \"WikipediaEntity\" %} \nAccording to wikipedia,\
+      \ {{question}} ||| {{answer.aliases|choice}} \n{% endif %}"
+    name: wikipedia_contextual_question
+    reference: Ask question using wikipedia keyword
+    task_template: true
+  5946db1a-a068-4a31-a06f-74a7d976cb6d: !Template
+    id: 5946db1a-a068-4a31-a06f-74a7d976cb6d
+    jinja: Guess a question from the topic "{{answer.aliases|choice}} " ||| {{question}}
```
Comment (VictorSanh) [inline templates/trivia_qa/rc/templates.yaml:38]:
```suggestion
    jinja: Guess a question from the topic "{{answer.aliases|choice}}" |||{{question}}
```

File: templates/trivia_qa/rc/templates.yaml [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    task_template: true
+  bfec3d73-c024-492f-8878-64fdb6639a29: !Template
+    id: bfec3d73-c024-492f-8878-64fdb6639a29
+    jinja: 'I''ve always wondered: {{question}} ||| {{answer.aliases|choice}} '
+    name: context_self_description
+    reference: Ask a question by self self description
+    task_template: true
+  c29c7072-0535-4e38-ba0c-b7ac0acdacf8: !Template
+    id: c29c7072-0535-4e38-ba0c-b7ac0acdacf8
+    jinja: 'Question : {{question}} {{"\n"}} Answer : ||| {{answer.aliases|choice}} '
```
Comment (VictorSanh) [inline templates/trivia_qa/rc/templates.yaml:88]:
```suggestion
    jinja: 'Question : {{question}} \nAnswer: |||{{answer.aliases|choice}} '
```

File: templates/trivia_qa/rc/templates.yaml (hunk 3 above)
Comment (VictorSanh) [reply templates/trivia_qa/rc/templates.yaml:11]: same for `question_multiple_valid_answer`

File: templates/trivia_qa/rc/templates.yaml [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    id: 69b1d0d7-ee22-4313-a8a0-89b718177d2a
+    jinja: "{% if answer.type == \"WikipediaEntity\" %} \nI''ve always wondered: {{question}}{{\"\
+      \\n\"}}\nCan you search wikipedia and answer? ||| {{answer.aliases|choice}}\
+      \ \n{% endif %}\n"
+    name: context_self_description_wikipedia
+    reference: Ask question by self description from wikipedia
+    task_template: true
+  91d9f950-a25a-4557-a16f-952d74629584: !Template
+    id: 91d9f950-a25a-4557-a16f-952d74629584
+    jinja: 'Answer the following question. {{"\n"}} {{question}} ||| {{answer.aliases|choice}} '
```
Comment (VictorSanh) [inline templates/trivia_qa/rc/templates.yaml:52]:
```suggestion
    jinja: 'Answer the following question. \n{{question}} |||{{answer.aliases|choice}} '
```

File: templates/trivia_qa/rc/templates.yaml (hunk 3 above)
Comment (VictorSanh) [reply templates/trivia_qa/rc/templates.yaml:11]: same for `question_with_instruction_multiple_valid_answer`

File: templates/trivia_qa/rc/templates.yaml (hunk 2 above)
Comment (VictorSanh) [reply templates/trivia_qa/rc/templates.yaml:79]: same for `wikipedia_contextual_question`

File: templates/trivia_qa/rc/templates.yaml (hunk 3 above)
Comment (VictorSanh) [reply templates/trivia_qa/rc/templates.yaml:11]: and `wikipedia_contextual_question_multiple_answer`

File: templates/trivia_qa/rc/templates.yaml (hunk 2 above)
Comment (VictorSanh) [reply templates/trivia_qa/rc/templates.yaml:79]: same for `wikipedia_question`

File: templates/trivia_qa/rc/templates.yaml (hunk 3 above)
Comment (VictorSanh) [reply templates/trivia_qa/rc/templates.yaml:11]: and `wikipedia_question_multiple_answer`

**Review by VictorSanh — CHANGES_REQUESTED** [review]
thank you @sbmaruf , please see my comments

TriviaQA is initially intended as a reading comprehension task (not a closed-book qa task), so:
- please correct the `task_template` flags
- add templates that actually leverage the multiple context provided (I am not sure which one should be used but there is at least `entity_pages.wiki_context`, `search_results.description`,  and `search_results.search_context`)

Also, use `choice` instead of listing of all (potentially repetitive) options

**Comment by sbmaruf** [comment]
@VictorSanh Can you take a look? Let me know if you find any issues.

**Review by VictorSanh — APPROVED** [review]
thank you @sbmaruf , please update the spreadsheet, merging now

**Pull request merged**